demos/slack-alert-app/slackbot.py

```diff
@@ -0,0 +1,116 @@
+import time
+import logging
+import os
+import schedule
+
+import pxapi
+from slack.web.client import WebClient
+from slack.errors import SlackApiError
+
+
+# Pixie PxL script.
+PXL_SCRIPT = """
+import px
+
+df = px.DataFrame(table='http_events', start_time='-5m')
+
+# Filter for px-sock-shop namespace only.
+df.namespace = df.ctx['namespace']
+df = df[df.namespace == 'px-sock-shop']
+
+# Add column for service name.
+df.service = df.ctx['service']
+
+# Add olumn for HTTP response status errors.
+df.error = df.http_resp_status >= 400
+
+# Group HTTP events by service, counting errors and total HTTP events.
+df = df.groupby(['service']).agg(
+    error_count=('error', px.sum),
+    total_requests=('http_resp_status', px.count)
+)
+
+px.display(df, "status")
+"""
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+def getPixieData(api_key, cluster_id):
+    # Get data from the Pixie API.
+
+    # Create a Pixie client.
+    logging.debug("Authorizing Pixie client.")
+    px_client = pxapi.Client(token=api_key)
+
+    # Connect to cluster.
+    conn = px_client.connect_to_cluster(cluster_id)
+    logging.debug("Pixie client connected to cluster.")
+
+    # Execute the PxL script.
+    script = conn.prepare_script(PXL_SCRIPT)
+    logging.debug("Pixie cluster executed script.")
+
+    service_stats_msg = ["*Recent 4xx+ Spikes in last 5 minutes:*"]
+
+    # Process table output rows.
+    for row in script.results("status"):
+        service_stats_msg.append(formatMessage(row["service"],
+                                               row["total_requests"],
+                                               row["error_count"]))
+
+    return "\n\n".join(service_stats_msg)
+
+
+def formatMessage(service, request_count, error_count):
+    # Format Pixie API table row data.
+    print(service, " ---> ", error_count,
```

Comment:
use logging here if you're gonna use it package wide

Reply:
actually seems unnecessary here. If you need to log this, log it in whatever consumes `formatMessage`

Reply:
This print line wasn't supposed to still be included.